servicetalk-http-netty/src/main/java/io/servicetalk/http/netty/SniCompleteChannelSingle.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright © 2021 Apple Inc. and the ServiceTalk project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.servicetalk.http.netty;
+
+import io.servicetalk.concurrent.SingleSource;
+import io.servicetalk.transport.netty.internal.ChannelInitializer;
+
+import io.netty.channel.Channel;
+import io.netty.channel.ChannelHandler;
+import io.netty.channel.ChannelHandlerContext;
+import io.netty.channel.ChannelInboundHandlerAdapter;
+import io.netty.handler.ssl.SniCompletionEvent;
+
+import java.nio.channels.ClosedChannelException;
+import javax.annotation.Nullable;
+
+import static io.servicetalk.transport.netty.internal.ChannelCloseUtils.assignConnectionError;
+
+final class SniCompleteChannelSingle extends ChannelInitSingle<SniCompletionEvent> {
+    private final boolean forceRead;
```

Comment:
This flag is always true in the current usage. Is it necessary? If so, a test demonstrating the need for a false variant would be useful.

Reply:
good point, I will remove this flag. This was taken from `AlpnChannelSingle` which is used on the client (doesn't need force read) and server (needs force read), however in this case we only need to cover the server use case.